src/Compilers/CSharp/Portable/Lowering/Instrumentation/DebugInfoInjector.cs

```diff
@@ -58,21 +58,65 @@ public override BoundStatement InstrumentExpressionStatement(BoundExpressionStat
+
+                    case ConstructorInitializerSyntax ctorInit:
+                        // Explicit constructor initializer:
+                        Debug.Assert(ctorInit.Parent is ConstructorDeclarationSyntax);
+                        return new BoundSequencePointWithSpan(ctorInit, rewritten, createSpanForConstructorInitializer((ConstructorDeclarationSyntax)ctorInit.Parent));
+
+                    case TypeDeclarationSyntax typeDecl:
+                        // Primary constructor with implicit base initializer:
+                        //   class [|C<T>(int X, int Y)|] : B;
+                        Debug.Assert(typeDecl.ParameterList != null);
+                        Debug.Assert(typeDecl.BaseList?.Types.Any(t => t is PrimaryConstructorBaseTypeSyntax { ArgumentList: not null }) != true);
+
+                        return new BoundSequencePointWithSpan(typeDecl, rewritten, TextSpan.FromBounds(typeDecl.Identifier.SpanStart, typeDecl.ParameterList.Span.End));
+
+                    case PrimaryConstructorBaseTypeSyntax baseInit:
+                        // Explicit base initializer:
+                        //   class C<T>(int X, int Y) : [|B(...)|];
+                        return new BoundSequencePointWithSpan(baseInit, rewritten, baseInit.Span);
+
+                    default:
+                        throw ExceptionUtilities.UnexpectedValue(original.Syntax.Kind());
                 }
             }
+            else if (original.Syntax is ParameterSyntax parameterSyntax)
+            {
+                // Record property setter sequence point.
+                return new BoundSequencePointWithSpan(parameterSyntax, rewritten, CreateSpan(parameterSyntax));
+            }
 
             return AddSequencePoint(rewritten);
+
+            static TextSpan createSpanForConstructorInitializer(ConstructorDeclarationSyntax constructorSyntax)
+            {
+                if (constructorSyntax.Initializer != null)
+                {
+                    //  [SomeAttribute] public MyCtorName(params int[] values): [|base()|] { ... }
+                    var start = constructorSyntax.Initializer.ThisOrBaseKeyword.SpanStart;
+                    var end = constructorSyntax.Initializer.ArgumentList.CloseParenToken.Span.End;
+                    return TextSpan.FromBounds(start, end);
+                }
+
+                if (constructorSyntax.Modifiers.Any(SyntaxKind.StaticKeyword))
+                {
+                    Debug.Assert(constructorSyntax.Body != null);
+
+                    // [SomeAttribute] static MyCtorName(...) [|{|] ... }
+                    var start = constructorSyntax.Body.OpenBraceToken.SpanStart;
+                    var end = constructorSyntax.Body.OpenBraceToken.Span.End;
+                    return TextSpan.FromBounds(start, end);
+                }
+
+                //  [SomeAttribute] [|public MyCtorName(params int[] values)|] { ... }
```

Comment:
nit: why include the modifiers?

Reply:
The span has always included them, so not changing the current behavior for constructors.

I think the reason is that usually modifiers are at the beginning of the line, so it's more intuitive to start the span there.